src/creature_instances.c

```diff
@@ -56,19 +57,19 @@ extern "C" {
 // DLLIMPORT struct InstanceInfo _DK_instance_info[48];
 // #define instance_info _DK_instance_info
 /******************************************************************************/
-long instf_attack_room_slab(struct Thing *creatng, long *param);
-long instf_creature_cast_spell(struct Thing *creatng, long *param);
-long instf_creature_fire_shot(struct Thing *creatng, long *param);
-long instf_damage_wall(struct Thing *creatng, long *param);
-long instf_destroy(struct Thing *creatng, long *param);
-long instf_dig(struct Thing *creatng, long *param);
-long instf_eat(struct Thing *creatng, long *param);
-long instf_fart(struct Thing *creatng, long *param);
-long instf_first_person_do_imp_task(struct Thing *creatng, long *param);
-long instf_pretty_path(struct Thing *creatng, long *param);
-long instf_reinforce(struct Thing *creatng, long *param);
-long instf_tortured(struct Thing *creatng, long *param);
-long instf_tunnel(struct Thing *creatng, long *param);
+long instf_attack_room_slab(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_creature_cast_spell(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_creature_fire_shot(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_damage_wall(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_destroy(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_dig(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_eat(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_fart(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_first_person_do_imp_task(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_pretty_path(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_reinforce(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_tortured(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
+long instf_tunnel(struct Thing *creatng, long *param, PlayerNumber plyr_idx);
 
```

Comment:
is it needed to pass the PlayerNumber at this level? can always be determined by the creatng you're also sending right?

Reply:
From what I can tell, all these functions must have the same number of parameters.

Reply:
yeah I meant for all of them

Reply:
They need the parameter, or the compiler will complain of too few parameters.